### PR TITLE
Fix: double deprecated calls when already refactored

### PR DESCRIPTION
**Pull request by bbrala**
## Description
This will fix the double deprecated calls. In the future we might need to refactor this, since something might actually need a nested deprecated call. That would be kinda nasty, but is an edge case that could happen (tm).

Codestyle is a bit noisy, annoying, but ah well.

## To Test
Run test.

## Drupal.org issue
https://www.drupal.org/project/rector/issues/3414169

### CODE REVIEW

File: src/Rector/AbstractDrupalCoreRector.php [hunk 1]
```diff
@@ -29,6 +31,31 @@ public function configure(array $configuration): void
...
+            return false;
+        }
+        if ($function->getName() !== 'backwardsCompatibleCall'
+            || $function->getDeclaringClass()->getName() !== DeprecationHelper::class
+        ) {
+            return false;
+        }
+
+        return $parameter !== null && $parameter->getName() === 'deprecatedCallable';
+    }
```
Comment (mglaman) [inline src/Rector/AbstractDrupalCoreRector.php:57]: Does Rector have a node visitor to just stop processing so that it doesn't handle nodes in this execution tree?

Comment (bbrala) [reply src/Rector/AbstractDrupalCoreRector.php:57]: Think so yes. My old fix for phpstan would probably also work. This seemed easier is 0retty much why I used this.

Comment (mglaman) [reply src/Rector/AbstractDrupalCoreRector.php:57]: I guess we're calling it in our base class, which all rules should extend. This is fine, but could lead to a "bug" if a rule didn't. But probably a minimal problem

**Review by mglaman — APPROVED** [review]

File: src/Rector/AbstractDrupalCoreRector.php (hunk 1 above)
Comment (bbrala) [reply src/Rector/AbstractDrupalCoreRector.php:57]: valid point. But if it doesn't extend this class, it should not add the deprecation helper. But you are right though, it could slip through if we need some sort of exotic way to do things. Still would probably use the base class for the helpers.
